.github/workflows/tpu-tests.yml

```diff
@@ -137,8 +116,30 @@ jobs:
           # failed run, print everything
           cat /tmp/full_output.txt;
         fi
-        exit $status_code
+
+        failed=$(kubectl get job $job_name -o jsonpath={.status.failed})
+        if [ -z "$failed" ] || [ $failed -eq 1 ]; then
+          echo "Error: the job failed"
+          kubectl get job $job_name -o jsonpath={.status}
+          kubectl get pod $pod_name -o jsonpath={.status}
+          exit 1
+        fi
+      shell: bash
+
+    - name: Cleanup
+      if: always()
       shell: bash
+      run: |
+        pod_name="${{ steps.deploy.outputs.pod_name }}"
+        job_name="${{ steps.deploy.outputs.job_name }}"
+        echo "Running cleanup: job $job_name, pod $pod_name"
+        # get events for the current pod
+        kubectl get event --field-selector involvedObject.name=$pod_name
+        # delete the job, this will also delete the associated pod
+        kubectl delete job $job_name
+        # show all resources
+        kubectl get pods
+        kubectl get jobs
```

Comment:
This will show what's left. Pods and jobs from this PR should've been deleted in the `kubectl delete` call above